backend/fms_core/services/sample_next_step.py

```diff
@@ -48,4 +48,96 @@ def queue_sample_to_study_workflow(sample_obj: Sample, study_obj: Study, order:
                                                              study=study_obj)
         except Exception as err:
             errors.append(err)
-    return sample_next_step, errors, warnings
\ No newline at end of file
+    return sample_next_step, errors, warnings
+
+def dequeue_sample_from_study_workflow(sample_obj: Sample, study_obj: Study, order: int=None) -> Tuple[Union[int, None], List[str], List[str]]:
```

Comment:
I don't like the way it could do either a single dequeue or a multiple one. I would create a different service for each. Also we have to keep in mind that finalized sample will have a next_step_step = None. So if you want to remove just that one you would end up removing all sample_next_step. This may change (we may want to store the workflow history of a sample in a more permanent way.

Reply:
I don't understand why would you end up removing all sample_next_step instances when finalized samples get to the last step ?

Reply:
If you specifically want to remove that sample_next_step ... it's step_order is None but providing an empty order to the service will be understood as remove all sample_next_step...